nixos/modules/services/x11/display-managers/default.nix

```diff
@@ -72,7 +72,7 @@ let
       sessionType="$1"
       if [ "$sessionType" = default ]; then sessionType=""; fi
 
-      ${optionalString (!cfg.displayManager.job.logsXsession && !cfg.displayManager.logToJournal) ''
+      ${optionalString cfg.displayManager.job.logsXsession ''
```

Comment:
Hmm, in the case where both `logsXsession` and `logToJournal` are `true` wouldn't you need to both redirect stdout/stderr to the file and pass them on to the journal? Something like
```bash
exec > >(tee -i ~/.xsession-errors) 2> >(tee -i /dev/stderr >> ~/.xsession-errors)
```
Disclamer: I don't know if the above is correct.

Reply:
@rycee First, thank you for the review. Yes, you are correct.
I tried `exec > >(tee -i ~/.xsession-errors) 2> >(tee -i ~/.xsession-errors >&2)` and seems to work.
The problem is that the script is executed multiple times and worse, if both `logXsession` and `logToJournal` are false the script is not executed at all.

I'm not sure how to deal with this.